src/models/__tests__/activation.test.ts

```diff
@@ -0,0 +1,224 @@
+import * as E from "fp-ts/Either";
+import * as O from "fp-ts/Option";
+
+import { Container } from "@azure/cosmos";
+import { NonNegativeInteger } from "@pagopa/ts-commons/lib/numbers";
+import { FiscalCode, NonEmptyString } from "@pagopa/ts-commons/lib/strings";
+import { ActivationStatusEnum } from "../../../generated/definitions/ActivationStatus";
+import { generateComposedVersionedModelId } from "../../utils/cosmosdb_model_composed_versioned";
+import { Activation, ActivationModel, RetrievedActivation, ACTIVATION_REFERENCE_ID_FIELD, ACTIVATION_MODEL_PK_FIELD, NewActivation,  } from "../activation";
+const aFiscalCode = "FRLFRC74E04B157I" as FiscalCode;
+const aServiceId = "xyz" as NonEmptyString;
+const aRawActivation: Activation = {
+  status: ActivationStatusEnum.ACTIVE,
+  fiscalCode: aFiscalCode,
+  serviceId: aServiceId
+};
+const aFirstVersionId = generateComposedVersionedModelId<Activation, typeof ACTIVATION_REFERENCE_ID_FIELD, typeof ACTIVATION_MODEL_PK_FIELD>(aServiceId, aFiscalCode, 0 as NonNegativeInteger);
+const aRetrivedActivation: RetrievedActivation = {
+  _etag: "_etag",
+  _rid: "_rid",
+  _self: "_self",
+  _ts: 1,
+  id: aFirstVersionId,
+  kind: "IRetrievedActivation",
+  version: 0 as NonNegativeInteger,
+  ...aRawActivation
+};
+
+describe("findLastVersionByModelId", () => {
+  it("should resolve to an existing activation", async () => {
+    const containerMock = ({
```

Comment:
I see this container mocked in every test. Maybe we could extract it and mock just the functions.

Reply:
Fixed in 31b83dd